features/flows/elastic/src/main/java/org/opennms/netmgt/flows/elastic/ElasticFlowRepository.java

```diff
@@ -124,16 +122,19 @@ public ElasticFlowRepository(MetricRegistry metricRegistry, JestClient jestClien
     }
 
     @Override
-    public void persistNetFlow5Packets(Collection<? extends NF5Packet> packets, FlowSource source) throws FlowException {
-        LOG.debug("Converting {} Netflow 5 packets from {} to flow documents.", packets.size(), source);
-        final List<FlowDocument> flowDocuments = packets.stream()
+    public <P> void persist(final Collection<? extends P> packets, final FlowSource source, final Converter<P> converter) throws FlowException {
+        LOG.debug("Converting {} flow packets from {} to flow documents.", packets.size(), source);
+        final Stream<FlowDocument> documents = packets.stream()
                 .map(converter::convert)
                 .flatMap(Collection::stream)
-                .collect(Collectors.toList());
-        enrichAndPersistFlows(flowDocuments, source);
+                .map(FlowDocument::from);
+        enrichAndPersistFlows(documents, source);
     }
 
-    public void enrichAndPersistFlows(List<FlowDocument> flowDocuments, FlowSource source) throws FlowException {
+    public void enrichAndPersistFlows(final Stream<FlowDocument> documents, FlowSource source) throws FlowException {
+        // TODO: Use streams here?
```

Comment:
Please own the TODO or remove it. If you want to own it: `// TODO DF: Use streams here?`